modules/javafx.graphics/src/main/java/com/sun/javafx/css/media/MediaFeatures.java

```diff
@@ -105,8 +213,46 @@ private static <T extends Enum<T>> T enumValue(Function<String, T> func, String
         try {
             return func.apply(featureValue.toUpperCase(Locale.ROOT));
         } catch (IllegalArgumentException e) {
-            throw new IllegalArgumentException(
-                String.format("Unknown value <%s> for media feature <%s>", featureValue, featureName));
+            throw unknownValue(featureName, featureValue);
+        }
+    }
+
+    private static String lowerCaseTextValue(String text) {
+        return text != null ? text.toLowerCase(Locale.ROOT).intern() : null;
+    }
+
+    private static Size lengthValue(String featureName, String lowerCaseText) {
+        int unitIndex = -1;
+
+        for (int i = 0; i < lowerCaseText.length(); i++) {
+            if (!Character.isDigit(lowerCaseText.charAt(i))) {
+                unitIndex = i;
```

Comment:
I think this is the place where the parser fails with fractional numbers...

Reply:
Yes. It turns out that we need to differentiate `<size>` and `<number>` values, as the latter don't have a unit (and can never have one). I've refactored the code a bit to handle both types of ranges.